docs/sorting-algorithms/selection-sort.mdx

```diff
@@ -0,0 +1,157 @@
+---
+title: Selection Sort
+slug: /sorting-algorithms/selection-sort
+---
+
+## Introduction
+
+Selection sort is a sorting algorithm that finds out the smallest element in the unsorted part in each iteration and puts it to the last of the sorted part. It does less memory writes as compared to other sorting algorithms. It is not stable and In place sorting algorithm. It takes the same time in every case `(best, average, worst)`.
+
+## Explanation 
+
+First, we will find out the smallest element in the range `[0,n-1]` and swap the smallest element with the element at index `0`. Then we will find the smallest element in the range `[1,n-1]` and swap it will the element at index `1` and we keep on reducing the range and swapping the smallest element with the first element in the range till the array gets sorted.
+
+Let's have one example: 
+```
+arr: 5 4 3 2 1
+
+Find the minimum element in arr[0...4]
+and swap it with arr[0]
+
+arr: 1 4 3 2 5 
+
+Find the minimum element in arr[1...4]
+and swap it with arr[1]
+
+arr: 1 2 3 4 5 
+
+Find the minimum element in arr[2...4]
+and swap it with arr[2]
+
+arr: 1 2 3 4 5 
+
+Find the minimum element in arr[3...4]
+and swap it with arr[3]
+```
+## Algorithm
+
+In Selection Sort, we follow the following algorithm:
+
+- First we will find the smallest element in the range `[0,n-1]` and swap it with the element at `0th` index
+
+- Then we will find the smallest element int the range `[1,n-1]` and swap it with the element at `1st` index
+
+- We keep on increasing the left part of the range until it reaches `n-1` and do the above mentioned process
+
+## Code
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+  ]
+}>
+<TabItem value="cpp">
+
+```cpp
+#include<bits/stdc++.h>
+using namespace std;
+
+void selection_sort(vector <int> &arr, int n) 
+{
+   int i=0;
+   while(i<n)
+   {
+      int mini=i;
+      int j=i+1;
+      while(j<n)
+      {
+         if(arr[j]<arr[mini])mini=j;
+         j++;
+      }
+      swap(arr[mini],arr[i]);
+      i++;
+   }   
+}
+
+int main() {
+   int n;
+   cout << "Enter the size of array:";
+   cin >> n;
+   vector <int> arr(n);
```

Comment:
Add a console statement that then asks the user to input the values as well

Reply:
done